include/envoy/buffer/buffer.h

```diff
@@ -102,6 +102,7 @@ class Instance {
   /**
    * Commit a set of slices originally obtained from reserve(). The number of slices can be
    * different from the number obtained from reserve(). The size of each slice can also be altered.
+   * Commit must occur following a reserve() without any mutating operations in between.
```

Comment:
It's valid (and even common) to change the `len_` field in a reserved slice between `reserve()` and `commit()`, just not the `mem_` field.